docs/functionaldatatypes/equivalence.md

```diff
@@ -0,0 +1,80 @@
+---
+id: equivalence
+title: "Equivalence"
+---
+
+An `Equivalence[A, B]` describes an equivalence relationship between two types `A` and `B`.
+
+An equivalence relationship is defined in terms of two functions `to` and `from` that convert a value of type `A` to a value of type `B` and vice versa.
+
+```scala mdoc
+case class Equivalence[A, B](to: A => B, from: B => A)
+```
+
+To be a valid `Equivalence` the functions `to` and `from` must satisfy an identity law, which says that for any value of type `A`, if we transform it to a `B` using `to` and then back using `from` we get the same value. The same property must also apply for the reverse.
+
+Simple examples of an equivalence relationship would be between an `List[A]` and a `Chunk[A]`. We can transform any `List` into a `Chunk` using the `Chunk.fromIterable` operator and we can transform any `Chunk` into a `List` using the `toList` operator.
+
+```scala mdoc:reset
+import zio.Chunk
+import zio.prelude.Equivalence
+
+def listChunkEquivalence[A]: Equivalence[List[A], Chunk[A]] =
+  Equivalence(Chunk.fromIterable, _.toList)
+```
+
+This essentially represents the fact that `List` and `Chunk` contain the same information. They represent it in different ways internally and have different performance characteristics but both of them model zero or more values of some type `A`.
+
+We can see from the example above that the order of describing `List` and `Chunk` in the equivalence relationship was somewhat arbitrary. We could just as well have described this as an equivalence between `Chunk[A]` and `List[A]`.
+
+The `Equivalence` data type lets us express that through the `flip` operator.
+
+```scala mdoc
+def chunkListEquivalence[A]: Equivalence[Chunk[A], List[A]] =
+  listChunkEquivalence.flip
+```
+
+In addition to being able to reverse equivalence relationships we can also compose them. If `A` is equivalent to `B` and `B` is equivalent to `C` then `A` is equivalent to `C`.
+
+We can express this using the `andThen` operator on `Equivalence` or its symbolic alias `>>>`.
+
+To demonstrate this let's define another equivalence relationship `Vector[A]` and `List[A]`.
+
+```scala mdoc
+def vectorListEquivalence[A]: Equivalence[Vector[A], List[A]] =
+  Equivalence(_.toList, _.toVector)
+```
+
+Given these two equivalence relationships we can then define an equivalence relationship between a `Vector[A]` and a `Chunk[A]`.
+
+```scala mdoc
+def vectorChunkEquivelence[A]: Equivalence[Vector[A], Chunk[A]] =
```

Comment:
typo a/e: you wrote `vectorChunkEquivelence` should be `...valence`

Reply:
Fixed.